### ISSUE TITLE
Introduce reactive global store(s)

### ISSUE THREAD

**Issue by maninak** [cytechmobile/radicle-vscode-extension#86]
As our logic gets more interdependent, UI depends more on state and different UIs depend on the same state, there's an increasingly higher need for a global store handling all of those concerns.

In such a store(s) we would declaratively define that C depends on A and B if there's something like `const C = A + B` in our logic and if either of those dependencies changes then C should be automatically recalculated further propagating effects upwards as needed. 

In essence, it would be _reactively_ calling procedures as dependencies change instead of us having to make sure that we are procedurally launching them. We've been doing without such a store 'till now on purpose (KISS) but continuing on without one will end up in more and more interwoven spaghetti code that will be prone to regressions. IMO now is a good moment to tackle this.

Starting out, we should trace everything that depends on extension configs (and also everything that depends on those things, etc..) and map that state and interdependencies in global store(s).

The refactoring must be done with great care so as to not introduce regressions. Ideally, this is tackled after #35 is done and all affected flows are already covered by E2E tests.

**NOTES:**

- [pinia](https://pinia.vuejs.org/core-concepts/) looks to be a good and lean, no-fluff option

**Comment by maninak** [comment]
**Results:**

### 🚀 Enhancements

- **webview:** make webviews like the one for patch details reactively adapt their UI state when any extension state they depend on is updated

### 🩹 Fixes

- **patch-list:** show check-out state per patch item always reflecting git state. Previously a checked out patch would not have the associated checkmark denoting its state shown in the patches list unless a check out AND a list refresh was done.
- **patch-detail:** the buttons on patch detail webviews left open from a previous VS Code session that got restored will now work, same as with just opened webviews
- **webview:** make webview restoration across sessions more robust. Previously webview restoration would sometimes fail resulting in an empty panel.
- **config:** watch _user-defined_ path to Radicle CLI binary for changes too. Previously only the default paths per OS were being watched.
- **onboarding:** detect Radicle CLI binary installation change even if file or parent directory tree is missing on extension's initialization

### 💅 Refactors

- **webview:** restructure architecture for better type-safety, easier maintenance and make the logic independent of the specific single webview available for now

### 🏡 Chores

- **store:** further progress on migrating formerly procedural state management to a reactive, declarative paradigm. See Chores of v0.4.0 for more info.

**Demo:**


https://github.com/cytechmobile/radicle-vscode-extension/assets/7542054/6a6d6ca5-55da-4925-88d8-545cd3b1707e
